docs/core/PINs/PIN-15-Skip-as-Finished-and-Conditionals.md

```diff
@@ -0,0 +1,66 @@
+---
+title: 'PIN-15: Skip state refactor & conditionals'
+sidebarDepth: 0
+---
+
+# PIN 15: Skip state as Finished + Control Flow Conditional
+
+Date: February 24, 2020
+
+Author: Alex Goodman
+
+## Status
+
+Proposed
+
+## Context
+
+The [ControlFlow](https://docs.prefect.io/api/latest/tasks/control_flow.html) Tasks allow for users to branch conditionally within the execution graph, skipping some tasks and allowing execution of others. However, [back-to-back conditions](https://github.com/PrefectHQ/prefect/issues/2017) have proven to be a problem under the existing implementation. Specifically, the problem appears to be with the semantics with the “Skip” Prefect State: it is considered to be fundamentally a “Success” state. That is, it is not easy (if impossible) to determine why a task was skipped, and if ignoring that Skipped state is OK. This means that using `skip_on_upstream_skip` is not a good enough mechanism to implement back-to-back conditions and hints that more functionality is needed.
```

Comment:
Could you add one sentence to this to expand on the ambiguity of the Skip state: right now it represents both tasks that were never run as well as tasks that raise SKIP signals (which is implicit in your comment on "why a task was skipped" but for posterity I want to document that ambiguity in particular)?

In addition, could you add a line / second paragraph maybe about "TriggerFailed" being a subclass of Failed  and why that is not ideal (because failures propagate, `any_failed` triggers run, etc.)

Reply:
Good call; added more info in [b3a2e41](https://github.com/PrefectHQ/prefect/pull/2071/commits/b3a2e4170118c4a53296e1bd2e97de1587bf973b) to account for these points.